tla-pp/src/main/scala/at/forsyte/apalache/tla/pp/passes/ConfigurationPassImpl.scala

```diff
@@ -47,15 +47,24 @@ class ConfigurationPassImpl @Inject()(val options: WriteablePassOptions,
     */
   override def execute(): Boolean = {
     val currentModule = tlaModule.get
+    val relevantOptions = copyRelevantOptions()
     // try to read from the TLC configuration file
-    loadOptionsFromTlcConfig(currentModule)
-    setFallbackOptions()
+    loadOptionsFromTlcConfig(currentModule, relevantOptions)
+    setFallbackOptions(relevantOptions)
 
     // make sure that the required operators are defined
-    ensureDeclarationsArePresent(currentModule)
+    ensureDeclarationsArePresent(currentModule, relevantOptions)
+
+    // copy the relevant options back in options
+    for (name <- NormalizedNames.STANDARD_OPTION_NAMES) {
+      relevantOptions.get[Any]("checker", name)
+        .collect { case value => options.set("checker." + name, value) }
```

Comment:
This is identical to the body of `copyRelevantOptions`, pace a parameter or two. Would it make more sense to abstract into a single function that is called two different times with different arguments? I may be off the mark, but I'm finding the logic hard to follow here. Seems like a lot of stateful mutations going on for something that appears to be a mere matter of filtering out members of a collection and verifying that they have certain properties.

Reply:
That is a good catch. Fixed.